otsserver/backup.py

```diff
@@ -139,10 +140,10 @@ def write_disk_cache(self, chunk, bytes):
         cache_path = self.cache_path + '/' + chunk_path
         os.makedirs(cache_path, exist_ok=True)
         cache_file = cache_path + '/' + chunk_str
-        cache_file_tmp = cache_file + '.tmp'
-        with open(cache_file_tmp, 'wb') as fd:
-            fd.write(bytes)
-        os.rename(cache_file_tmp, cache_file)  # rename is atomic
+        temp_file = NamedTemporaryFile(delete=False)
+        temp_file.write(bytes)
+        temp_file.close()
+        os.rename(temp_file.name, cache_file)  # rename is atomic
```

Comment:
Have you thought through what happens if this fails?

Since it's atomic, I *think* it should be enough to simply ignore the error if rename fails due to a file already existing; the file should have the correct content.

Reply:
unless you run Windows, https://docs.python.org/3/library/os.html#os.rename but in that case you are screwed anyway. If could use os.replace if we want to have the same behaviour